Lib/test/test_functools.py

```diff
@@ -403,6 +403,32 @@ def test_attributes_unwritable(self):
         else:
             self.fail('partial object allowed __dict__ to be deleted')
 
+    def test_manually_adding_non_string_keyword(self):
+        p = self.partial(capture)
+        # Adding a non-string/unicode keyword to partial kwargs
+        p.keywords[1234] = 4321
```

Comment:
It would be better to test the value that have different repr and str. For example a string.